src/main/java/org/folio/dcb/service/impl/TransactionAuditServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package org.folio.dcb.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.dcb.domain.dto.DcbTransaction;
+import org.folio.dcb.domain.entity.TransactionAuditEntity;
+import org.folio.dcb.domain.entity.TransactionEntity;
+import org.folio.dcb.domain.mapper.TransactionMapper;
+import org.folio.dcb.repository.TransactionAuditRepository;
+import org.folio.dcb.service.TransactionAuditService;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class TransactionAuditServiceImpl implements TransactionAuditService {
+  private static final String ERROR_ACTION = "ERROR";
+  private static final String DUPLICATE_ERROR_ACTION = "DUPLICATE_ERROR";
+  private static final String DUPLICATE_ERROR_TRANSACTION_ID = "-1";
+
+  private final TransactionMapper transactionMapper;
+  private final TransactionAuditRepository transactionAuditRepository;
+  @Override
+  public void logErrorIfTransactionAuditExists(String dcbTransactionId, String errorMsg) {
+    log.debug("logTheErrorForExistedTransactionAudit:: dcbTransactionId = {}, err = {}", dcbTransactionId, errorMsg);
+    TransactionAuditEntity auditExisting = transactionAuditRepository.findLatestTransactionAuditEntityByDcbTransactionId(dcbTransactionId).orElse(null);
+
+    if(auditExisting != null){
+      TransactionAuditEntity auditError = generateTrnAuditEntityFromTheFoundOneWithError(auditExisting, errorMsg);
+      transactionAuditRepository.save(auditError);
+    }
+  }
+
+  /**
+   * For the case the error happens during DCB transaction creation.
+   * At this time there is no transaction_audit data existed, which refers to current DCB transaction.
```

Comment:
I would say "persisted" rather than existed here and below

Reply:
fixed